crates/subspace-networking/Cargo.toml

```diff
@@ -20,14 +20,17 @@ async-trait = "0.1.53"
 bytes = "1.1.0"
 event-listener-primitives = "2.0.1"
 futures = "0.3.21"
+generic-array = "0.14.5"
 hex = "0.4.3"
 nohash-hasher = "0.2.0"
 parity-scale-codec = "3.1.2"
 parking_lot = "0.12.0"
+sha2 = "0.10.2"
```

Comment:
We have a `crypto::sha256_hash` for this in `subspace-core-primitives` that you already use in some places and that uses hardware acceleration for Sha256 among other things. It would be preferable to switch to that instead of using `sha2` crate directly.